src/uct/tcp/tcp_ep.c

```diff
@@ -730,26 +746,28 @@ ucs_status_t uct_tcp_ep_set_dest_addr(const uct_device_addr_t *dev_addr,
                                       const uct_iface_addr_t *iface_addr,
                                       struct sockaddr *dest_addr)
 {
-    uct_tcp_device_addr_t *tcp_dev_addr  = (uct_tcp_device_addr_t*)dev_addr;
-    uct_tcp_iface_addr_t *tcp_iface_addr = (uct_tcp_iface_addr_t*)iface_addr;
-    const struct in_addr loopback_addr   = {
+    uct_tcp_device_addr_t *tcp_dev_addr   = (uct_tcp_device_addr_t*)dev_addr;
+    uct_tcp_iface_addr_t *tcp_iface_addr  = (uct_tcp_iface_addr_t*)iface_addr;
+    const struct in_addr in4addr_loopback = {
         .s_addr = htonl(INADDR_LOOPBACK)
     };
     const void *in_addr;
     ucs_status_t status;
 
-    memset(dest_addr, 0, sizeof(*dest_addr));
+    memset(dest_addr, 0,
+           ((tcp_dev_addr->sa_family == AF_INET) ?
+                    sizeof(struct sockaddr_in) :
+                    sizeof(struct sockaddr_in6)));
+    dest_addr->sa_family = tcp_dev_addr->sa_family;
 
     if (tcp_dev_addr->flags & UCT_TCP_DEVICE_ADDR_FLAG_LOOPBACK) {
-        in_addr = &loopback_addr;
+        in_addr = (dest_addr->sa_family == AF_INET) ? (void*)&in4addr_loopback :
```

Comment:
does it meet the "80 symbols per line" rule?

Reply:
yes